packages/modal/src/index.ts

```diff
@@ -0,0 +1,24 @@
+import { OkModalAttribute } from './resources/ok-modal-attribute';
+import { DismissModalAttribute } from './resources/dismiss-modal-attribute';
+import { AttachFocusAttribute } from './resources/attach-focus-attribute';
+import { UxDefaultModalConfiguration } from './ux-modal-configuration';
+import { FrameworkConfiguration, PLATFORM } from 'aurelia-framework';
+
+export function configure(frameworkConfig: FrameworkConfiguration, callback?: (config: UxDefaultModalConfiguration) => void) {
+  frameworkConfig.globalResources([
+    AttachFocusAttribute,
```

Comment:
This will collide with anyone who uses `aurelia-dialog` together with this, is this suggesting that the modal service can cover all use cases of dialog?

Reply:
Good question. I believe it's very close but still I prefer to ensure it doesn't collide. First I wanted to name it differently (by prefixing something like `ux-modal-...` but then I realized that this attribute is, in fact, not so related to `aurelia-dialog` or `ux-modal` as it only call the element `focus()` api.

So my plan was to make it optional and let user opt-in/out for this attribute (as aurelia-dialog does as well). I can do it with a `useDefault()` pattern in the config class. Would that be ok ?

Thanks for catching this !

Reply:
I see, so maybe we can do this:
```ts
frameworkConfig.postTask(() => {
  const globalResources = frameworkConfig.container.get(ViewResources);
  if (globalResources.getAttribute('attach-focus') === null) {
    globalResources.registerAttribute(
      'attach-focus',
      metadata.getOwn(metadata.resource, AttachFocusAttribute)
    );
  }
})
```

Reply:
Sorry, i just realize this probably wont work, since the views will be compiled before the attribute is ready

Reply:
Ohh.. this is brilliant. Didn't know about `postTask()`. I tried to use the `registerAttribute()` method but failed to really understand it. I'm missing a third parameter to the method and I don't know how to build the `metadata` object ?

What if I do: 

```ts
  frameworkConfig.postTask(() => {
    const globalResources = frameworkConfig.aurelia.resources;
    if (globalResources.getAttribute('attach-focus') === null) {
      const extraConfig = new FrameworkConfiguration(frameworkConfig.aurelia);
      extraConfig.globalResources(AttachFocusAttribute);
      extraConfig.apply();
    }
  });

```

Reply:
Oh ok.. shame I liked it !

Reply:
`metadata` is an import from `aurelia-metadata`. For the 3rd parameter, I think we don't need it. Probably we will need to prefix it. Not the best, but consistently good I think

Reply:
ok, I'll go for the prefix version